napari/layers/source.py

```diff
@@ -0,0 +1,23 @@
+from typing import Optional
+
+from pydantic.dataclasses import dataclass
+
+
+@dataclass(frozen=True)
+class Source:
+    """An object to store the provenance of a layer.
+
+    Parameters
+    ----------
+    path: str, optional
+        filpath/url associated with layer
+    plugin: str, optional
+        name of plugin that loaded the file (if applicable)
```

Comment:
I think we should probably make this always be the reader plugin.

Reply:
What about a sample data plugin?  (Where no reader need be involved?)

Reply:
yup got it now, missed it above. this is fine